projects/core/src/global-message/store/effects/global-message.effect.spec.ts

```diff
@@ -84,4 +90,26 @@ describe('GlobalMessage Effects', () => {
       );
     });
   });
+  describe('removeDuplicated$', () => {
```

Comment:
Can you add a test case that checks if message are removed in the correct order?

Reply:
After a long investigation, we have come to the conclusion that we're unable to write that kind of unit test. Testing effects is simple and based on defining input and outcome. As far as I know, there is no option to handle checking it in the middle. Case have been covered in unit tests above, which tests removing strictly and always first element / message from list.